test_storage_s3_queue/test_4.py::test_registry is flaky

```
        registry, stat = zk.get(f"{keeper_path}/registry/")
    
        uuid2 = node1.query(
            f"SELECT uuid FROM system.tables WHERE database = '{db_name}' and table = '{table_name_2}'"
        ).strip()
    
        assert uuid1 in str(registry)
        assert uuid2 in str(registry)
    
        expected = [
            f"0\\ninstance\\n{uuid1}\\n",
            f"0\\ninstance2\\n{uuid1}\\n",
            f"0\\ninstance\\n{uuid2}\\n",
            f"0\\ninstance2\\n{uuid2}\\n",
        ]
    
        for elem in expected:
>           assert elem in str(registry)
E           assert '0\\ninstance2\\n796a56da-1148-4d77-9c11-6c175cdee378\\n' in "b'0\\ninstance\\n3095e8a7-e4e7-4e7e-947b-fa8258c8d1a3\\n,0\\ninstance2\\n3095e8a7-e4e7-4e7e-947b-fa8258c8d1a3\\n,0\\ninstance\\n796a56da-1148-4d77-9c11-6c175cdee378\\n'"
E            +  where "b'0\\ninstance\\n3095e8a7-e4e7-4e7e-947b-fa8258c8d1a3\\n,0\\ninstance2\\n3095e8a7-e4e7-4e7e-947b-fa8258c8d1a3\\n,0\\ninstance\\n796a56da-1148-4d77-9c11-6c175cdee378\\n'" = str(b'0\ninstance\n3095e8a7-e4e7-4e7e-947b-fa8258c8d1a3\n,0\ninstance2\n3095e8a7-e4e7-4e7e-947b-fa8258c8d1a3\n,0\ninstance\n796a56da-1148-4d77-9c11-6c175cdee378\n')
```

https://s3.amazonaws.com/clickhouse-test-reports/json.html?PR=78449&sha=25736fb9333701dab544cbc303a8d1ee0a1ea2c7&name_0=PR&name_1=Integration%20tests%20%28release%2C%201%2F4%29

--- a/tests/integration/test_storage_s3_queue/test_4.py
+++ b/tests/integration/test_storage_s3_queue/test_4.py
@@ -624,6 +624,10 @@ def test_registry(started_cluster):
         database_name=db_name,
     )
 
+    # ensure that the table is created on node2 before we query the registry as there might be a race between the time
+    # we actually create the table on node2 and when we query the registry.
+    node2.query(f"SYSTEM SYNC DATABASE REPLICA {db_name}")
+
     registry, stat = zk.get(f"{keeper_path}/registry/")
 
     uuid2 = node1.query(

--- a/tests/integration/test_storage_s3_queue/test_4.py
+++ b/tests/integration/test_storage_s3_queue/test_4.py
@@ -1,16 +1,10 @@
-import io
-import json
 import logging
-import random
-import string
 import time
 import uuid
 from multiprocessing.dummy import Pool
 
 import pytest
-from kazoo.exceptions import NoNodeError
 
-from helpers.client import QueryRuntimeException
 from helpers.cluster import ClickHouseCluster, ClickHouseInstance
 from helpers.s3_queue_common import (
     run_query,
@@ -140,7 +134,7 @@ def test_replicated(started_cluster):
         f"SELECT * FROM system.zookeeper WHERE path = '{keeper_path}'"
     )
 
-    total_values = generate_random_files(
+    generate_random_files(
         started_cluster, files_path, files_to_generate, start_ind=0, row_num=1
     )
 
@@ -170,10 +164,8 @@ def test_bad_settings(started_cluster):
     node = started_cluster.instances["node_cloud_mode"]
 
     table_name = f"test_bad_settings_{uuid.uuid4().hex[:8]}"
-    dst_table_name = f"{table_name}_dst"
     keeper_path = f"/clickhouse/test_{table_name}"
     files_path = f"{table_name}_data"
-    files_to_generate = 10
 
     try:
         create_table(
@@ -225,7 +217,7 @@ def test_processing_threads(started_cluster):
         )
     )
 
-    total_values = generate_random_files(
+    generate_random_files(
         started_cluster, files_path, files_to_generate, start_ind=0, row_num=1
     )
 
@@ -296,7 +288,7 @@ def test_alter_settings(started_cluster):
         f"SELECT * FROM system.zookeeper WHERE path = '{keeper_path}'"
     )
 
-    total_values = generate_random_files(
+    generate_random_files(
         started_cluster, files_path, files_to_generate, start_ind=0, row_num=1
     )
 
@@ -588,7 +580,7 @@ def test_registry(started_cluster):
     for elem in expected:
         assert elem in str(registry)
 
-    total_values = generate_random_files(
+    generate_random_files(
         started_cluster, files_path, files_to_generate, start_ind=0, row_num=1
     )
 

